packages/apps/stock/src/screens/internalMoves/InternalMoveCreationScreen.js

```diff
@@ -59,101 +94,113 @@ const InternalMoveCreationScreen = ({navigation}) => {
   const handleToStockLocationChange = useCallback(
     _value => {
       if (_value == null) {
-        handleReset(CREATION_STEP.toStockLocation);
+        handleReset(InternalMoveCreation.step.toStockLocation);
       } else {
         setToStockLocation(_value);
-        handleNextStep(CREATION_STEP.toStockLocation);
+        handleNextStep(InternalMoveCreation.step.toStockLocation);
       }
     },
     [handleNextStep, handleReset],
   );
 
-  const handleReset = useCallback((_step = CREATION_STEP.fromStockLocation) => {
-    setStep(_step);
+  const handleReset = useCallback(
+    (_step = InternalMoveCreation.step.fromStockLocation) => {
+      setCurrentStep(_step);
 
-    if (_step === CREATION_STEP.toStockLocation) {
-      setToStockLocation(null);
-    }
+      if (_step === InternalMoveCreation.step.fromStockLocation) {
+        setFromStockLocation(null);
+      }
 
-    if (_step === CREATION_STEP.fromStockLocation) {
-      setFromStockLocation(null);
-    }
-  }, []);
+      if (_step === InternalMoveCreation.step.addLine) {
+        setNewLine(null);
+      }
 
-  const handleNextStep = useCallback(
-    _current => {
-      setStep(() => {
-        if (_current === CREATION_STEP.fromStockLocation) {
-          if (toStockLocation != null) {
-            return CREATION_STEP.lines;
-          } else {
-            return CREATION_STEP.toStockLocation;
-          }
-        }
-        if (_current <= CREATION_STEP.toStockLocation) {
-          return CREATION_STEP.lines;
-        }
-        return _current;
-      });
+      if (_step === InternalMoveCreation.step.toStockLocation) {
+        setToStockLocation(null);
+      }
     },
-    [toStockLocation],
+    [],
   );
 
-  const handleCompleteLines = useCallback(() => {
-    navigation.navigate('InternalMoveLineCreationScreen', {
-      fromStockLocation,
-      toStockLocation,
-      notes,
+  const handleNextStep = useCallback(_current => {
+    setCurrentStep(() => {
+      if (_current === InternalMoveCreation.step.fromStockLocation) {
+        return InternalMoveCreation.step.addLine;
+      }
+      return _current;
     });
-  }, [fromStockLocation, navigation, notes, toStockLocation]);
+  }, []);
```

Comment:
Not really useful here, the only case where you are really using a logic to change step is from fromStockLocation to addLine which is only called one time in handleFromStockLocationChange, everywhere else you are giving the next value. You could just use the setter to do that.